app/com/linkedin/drelephant/exceptions/TonYExceptionFingerprinting.java

```diff
@@ -0,0 +1,278 @@
+        exceptionInfoList.add(exceptionInfo);
+      }
+    }
+    return exceptionInfoList;
+  }
+
+  public void saveExceptionFingerprintingData() {
+    if (_exceptionInfoList != null) {
+      final String NOT_APPLICABLE = "NA";
+      String exceptionsTrace = convertToJSON(_exceptionInfoList);
+
+      JobsExceptionFingerPrinting jobsExceptionFingerPrinting = new JobsExceptionFingerPrinting();
+      jobsExceptionFingerPrinting.appId = NOT_APPLICABLE;
+      jobsExceptionFingerPrinting.taskId = NOT_APPLICABLE;
+      jobsExceptionFingerPrinting.flowExecUrl = _appResult.flowExecUrl;
+      jobsExceptionFingerPrinting.jobName = getJobName(_appResult.jobExecUrl);
+      jobsExceptionFingerPrinting.exceptionLog = "";
+      jobsExceptionFingerPrinting.exceptionType = HadoopException.HadoopExceptionType.TONY.toString();
+
+      JobsExceptionFingerPrinting tonyJobException = new JobsExceptionFingerPrinting();
+      tonyJobException.flowExecUrl = _appResult.flowExecUrl;
+      tonyJobException.appId = _appResult.id;
+      tonyJobException.taskId = NOT_APPLICABLE;
+      tonyJobException.jobName = getJobName(_appResult.jobExecUrl);
+      tonyJobException.logSourceInformation = _analyticJob.getAmContainerLogsURL();
+      //sparkJobException.exceptionLog = exceptionsTrace;
+      if (exceptionsTrace.trim().length() > 2) {
+        tonyJobException.exceptionLog = exceptionsTrace;
+      } else {
+        tonyJobException.exceptionLog = "Couldn't gather driver logs for the job";
+      }
+      tonyJobException.exceptionType = ExceptionInfo.ExceptionSource.DRIVER.toString();
+      tonyJobException.exceptionLog = exceptionsTrace;
+      tonyJobException.save();
+      jobsExceptionFingerPrinting.save();
+    }
+  }
+
+  private String fetchLogDataFromAddress(String containerURL) {
+      try {
+        logger.info("Fetching log data from URL " + containerURL);
+        URL amAddress = new URL(containerURL);
+        HttpURLConnection connection = (HttpURLConnection) amAddress.openConnection();
+        connection.connect();
+        InputStream inputStream = connection.getInputStream();
+        String responseString = IOUtils.toString(inputStream);
+        String logDataString = filterOutLogsFromHTMLResponse(responseString);
+        logDataString = StringEscapeUtils.unescapeHtml4(logDataString);
+        //Todo: remove this before PR, this for implementation time debugging
+        logger.info("Length and Response Code for " + _appResult.id + " "  + responseString.length() + " " + connection.getResponseCode());
+        inputStream.close();
+        connection.disconnect();
+        return logDataString;
+      } catch (IOException ex) {
+        logger.error("Error occured while fetching log data from " + containerURL, ex);
+      }
+      return null;
+  }
+
+  private String filterOutLogsFromHTMLResponse(String htmlResponse)  {
+    String regex_for_pattern_of_logs_in_html_response = "<pre>([\\s\\S]+)</pre>";
```

Comment:
Same comment , move this up , at the declaration of the class ,

Reply:
Used it only once so, kept it as a local variable.